Builder/ParametersGenerator.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Becklyn\RouteTreeBundle\Builder;
+
+use Becklyn\RouteTreeBundle\Tree\Node;
+
+
+/**
+ *
+ */
+class ParametersGenerator
+{
+    /**
+     * Automatically sets the parameters for all descendant nodes
+     *
+     * @param Node $node
+     */
+    public function generateParametersForNode (Node $node)
+    {
+        $node->setParameters($this->calculateParametersForNode($node));
```

Comment:
If you take a look at line 38-42, we first load all existing parameters from the node and overwrite only the ones that are `null`. That means, that setting `null` explicitly doesn't work.